src/rpc/blockchain.cpp

```diff
@@ -490,21 +490,25 @@ static void entryToJSON(const CTxMemPool& pool, UniValue& info, const CTxMemPool
     fees.pushKV("modified", ValueFromAmount(e.GetModifiedFee()));
     fees.pushKV("ancestor", ValueFromAmount(e.GetModFeesWithAncestors()));
     fees.pushKV("descendant", ValueFromAmount(e.GetModFeesWithDescendants()));
-    info.pushKV("fees", fees);
-
     info.pushKV("vsize", (int)e.GetTxSize());
     info.pushKV("weight", (int)e.GetTxWeight());
-    info.pushKV("fee", ValueFromAmount(e.GetFee()));
-    info.pushKV("modifiedfee", ValueFromAmount(e.GetModifiedFee()));
+    // TODO: top-level fee fields are deprecated. IsDeprecatedRPCEnabled("fees") blocks should be removed in v24
+    if (IsDeprecatedRPCEnabled("fees")) {
+        info.pushKV("fee", ValueFromAmount(e.GetFee()));
+        info.pushKV("modifiedfee", ValueFromAmount(e.GetModifiedFee()));
+    }
     info.pushKV("time", count_seconds(e.GetTime()));
     info.pushKV("height", (int)e.GetHeight());
     info.pushKV("descendantcount", e.GetCountWithDescendants());
     info.pushKV("descendantsize", e.GetSizeWithDescendants());
-    info.pushKV("descendantfees", e.GetModFeesWithDescendants());
+    if (IsDeprecatedRPCEnabled("fees"))
+        info.pushKV("descendantfees", e.GetModFeesWithDescendants());
     info.pushKV("ancestorcount", e.GetCountWithAncestors());
     info.pushKV("ancestorsize", e.GetSizeWithAncestors());
-    info.pushKV("ancestorfees", e.GetModFeesWithAncestors());
+    if (IsDeprecatedRPCEnabled("fees"))
+        info.pushKV("ancestorfees", e.GetModFeesWithAncestors());
```

Comment:
- brackets per developer-notes.md:
```
  - If an `if` only has a single-statement `then`-clause, it can appear
    on the same line as the `if`, without braces. In every other case,
    braces are required, and the `then` and `else` clauses must appear
    correctly indented on a new line.
```
- `IsDeprecatedRPCEnabled()` loops through a vector to match strings, so maybe invoke it once and store the result

```diff
@@ -493,7 +493,8 @@ static void entryToJSON(const CTxMemPool& pool, UniValue& info, const CTxMemPool
     info.pushKV("vsize", (int)e.GetTxSize());
     info.pushKV("weight", (int)e.GetTxWeight());
     // TODO: top-level fee fields are deprecated. IsDeprecatedRPCEnabled("fees") blocks should be removed in v24
-    if (IsDeprecatedRPCEnabled("fees")) {
+    const bool deprecated_fees_enabled{IsDeprecatedRPCEnabled("fees")};
+    if (deprecated_fees_enabled) {
         info.pushKV("fee", ValueFromAmount(e.GetFee()));
         info.pushKV("modifiedfee", ValueFromAmount(e.GetModifiedFee()));
     }
@@ -501,12 +502,14 @@ static void entryToJSON(const CTxMemPool& pool, UniValue& info, const CTxMemPool
     info.pushKV("height", (int)e.GetHeight());
     info.pushKV("descendantcount", e.GetCountWithDescendants());
     info.pushKV("descendantsize", e.GetSizeWithDescendants());
-    if (IsDeprecatedRPCEnabled("fees"))
+    if (deprecated_fees_enabled) {
         info.pushKV("descendantfees", e.GetModFeesWithDescendants());
+    }
     info.pushKV("ancestorcount", e.GetCountWithAncestors());
     info.pushKV("ancestorsize", e.GetSizeWithAncestors());
-    if (IsDeprecatedRPCEnabled("fees"))
+    if (deprecated_fees_enabled) {
         info.pushKV("ancestorfees", e.GetModFeesWithAncestors());
+    }
```